src/main/java/com/gotocompany/depot/maxcompute/converter/payload/StructPayloadConverter.java

```diff
@@ -0,0 +1,32 @@
+package com.gotocompany.depot.maxcompute.converter.payload;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import com.google.protobuf.util.JsonFormat;
+import com.gotocompany.depot.maxcompute.converter.type.StructTypeInfoConverter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class StructPayloadConverter implements PayloadConverter {
+
+    private final StructTypeInfoConverter structTypeInfoConverter;
+    private final JsonFormat.Printer printer = JsonFormat.printer()
+            .preservingProtoFieldNames()
+            .omittingInsignificantWhitespace();
+
+    @Override
+    public Object convertSingular(Descriptors.FieldDescriptor fieldDescriptor, Object object) {
+        try {
+            return printer.print((Message) object);
+        } catch (InvalidProtocolBufferException e) {
+            return "";
+        }
+    }
```

Comment:
The struct conversion silently returns empty string on error, which could hide issues. Also, the JSON conversion doesn't handle large structures efficiently and could cause memory issues.

Reply:
Same handler for BQ implementation, this could be refactored when doing the json mc implementation